### ISSUE TITLE
Does this work?

### ISSUE THREAD

**Issue by remino** [csstools/postcss-sass#40]
I’m not sure I understand the purpose of this plugin. Can it take files both in Sass and SCSS formats and turn them into CSS? If so, that doesn’t seem to work.

First, the first example in README doesn’t work as `process` in undefined:

```js
require('@csstools/postcss-sass').process
// undefined
```

Second, trying to pass Sass strings to it just fails:

```
> postcss = require('postcss')
> postcss([require('@csstools/postcss-sass')()]).process(`\nbody\n  color:red\n`)
LazyResult {
  stringified: false,
  processed: true,
  error: CssSyntaxError: <css input>:3:3: Unknown word
```

Is there something I’m missing?

**Comment by romainmenke** [comment]
I haven't used this plugin myself, so I can't speak to the limits of what it can and cannot do. But it is expected to work.

see the readme for PostCSS : https://github.com/csstools/postcss-sass?tab=readme-ov-file#postcss

```
const postcss = require('postcss');
const postcssSass = require('@csstools/postcss-sass');

postcss([
  postcssSass(/* pluginOptions */)
]).process(YOUR_CSS, {
  syntax: require('postcss-scss')
});
```

The readme might be outdated and incorrect but that section seems to be correct.

**Comment by remino** [comment]
@romainmenke Thanks for your reply.

If you look at my code, you’ll see it does what the sample you posted from the README does.

But to better illustrate the problem, feel free to try my PoC repo I made for this issue:
https://github.com/remino/postcss-sass-sample

I even committed a copy of the output I get in `log.txt`. It seems that SCSS will work, but Sass does not.

**Comment by remino** [comment]
I investigated further. I cracked open `node_modules/sass/sass.dart.js` and shoved a `console.log` call at line 27127 in `render()`:

```diff
     render(options, callback) {
+        console.log('======================', options)
         var _0_0;
```

The result of that, as expected, passing the SCSS input will eventually output that `console.log` just fine. However, with Sass, PostCSS checks first with the `postcss-scss` parser to see if the input is valid. Since it’s Sass and not SCSS, the parser tells PostCSS the syntax is invalid and PostCSS complains about it. The `render()` in the `sass` engine isn’t even called.

Although I think the README could mention that as a warning, it doesn’t look like it’s a problem with this package.

I guess the best workaround I can come up with is simply not use this package and the parser at all and just use the `sass` package directly:

```
const YOUR_SASS = /* sass */`
  $color: red

  body
    color: $color
`

const postcss = require('postcss');
const sass = require('sass');

postcss().process(
  sass.compileString(YOUR_SASS, { syntax: 'indented' }).css,
  {
    from: '', // Added option to make PostCSS stop complaining
    to: '', // Added option to make PostCSS stop complaining
  },
)
  .then(result => { console.log(result.css) })
  .catch(console.error);
```

But then that may also a challenge when supporting multiple files. Anyway, it’s a start.
